test/e2e/framework/service/jig.go

```diff
@@ -1051,3 +958,121 @@ func (j *TestJig) CreateSCTPServiceWithPort(tweak func(svc *v1.Service), port in
 	}
 	return j.sanityCheckService(result, svc.Spec.Type)
 }
+
+// The following test functions build on top of the 'jig', but are still methods of it because so that
+// they can reuse some of the fixtures (i.e. the client).
+
+func (j *TestJig) testReachabilityOverServiceName(serviceName string, sp v1.ServicePort, execPod *v1.Pod) error {
+	return j.testEndpointReachability(serviceName, sp.Port, sp.Protocol, execPod)
+}
+
+func (j *TestJig) testReachabilityOverClusterIP(clusterIP string, sp v1.ServicePort, execPod *v1.Pod) error {
+	// If .spec.clusterIP is set to "" or "None" for service, ClusterIP is not created, so reachability can not be tested over clusterIP:servicePort
+	isClusterIPV46, err := regexp.MatchString(e2enetwork.RegexIPv4+"||"+e2enetwork.RegexIPv6, clusterIP)
```

Comment:
you can use `isInvalidOrLocalhostAddress()` or 	
```
parsedIP := net.ParseIP(ip)
	if parsedIP == nil 
```

Reply:
(move of the earlier code) but valid point i think

Reply:
Right now this file has 2 sets of functions:
- those that are mehods of TestJig
- those that are not methods of TestJig

Now, it turns out that  one of the reusable functions here, `TestServiceReachability`, *needs* a client (so that it can preverify pods and so on before failing an exec...as that way the use of the function has much more clear test faillure resutls).

So, it seems like, since these funcs are all in the same file (and not heavily relied on outside this file) we, can couple all the test functionality here *to* the testJig, so that all functions in this file can reuse the other TestJig functionality.

*alternative*

now maybe we could also consider splitting the "TestJig" functionality out if we want to, and we could manually plumb the client into the function .